### PR TITLE
Remove all MPI_COMM_WORLD from src

### PR DESCRIPTION
**Pull request by streeve**
Could conflict with other apps

### CODE REVIEW

**Comment by MattRolchigo** [comment]
Should we test this for some small coupled finch-exaca runs before merging to make sure it didn't break anything?

**Comment by streeve** [comment]
> Should we test this for some small coupled finch-exaca runs before merging to make sure it didn't break anything?

It breaks compile for the coupled ExaCA build, so we definitely need to decide how to handle this - could keep the old version and mark as deprecated?

**Comment by MattRolchigo** [comment]
Does it break compile because of the fact that ExaCA uses the hardcoded MPI_COMM_WORLD everywhere? I do like the idea of moving away from that, particularly for my dream of coupled Finch-ExaCA runs on M+N nodes where there's a Finch communicator for ranks on the M nodes running layer n+1 while ExaCA and the communicator for ranks on the N nodes runs layer n

**Comment by streeve** [comment]
> Does it break compile because of the fact that ExaCA uses the hardcoded MPI_COMM_WORLD everywhere? I do like the idea of moving away from that, particularly for my dream of coupled Finch-ExaCA runs on M+N nodes where there's a Finch communicator for ranks on the M nodes running layer n+1 while ExaCA and the communicator for ranks on the N nodes runs layer n

Only because we changed the API here to pass a communicator where before there were no args - but I share your dream. I'll do this the deprecated route to make life easier in ExaCA, then plan out how to rip out COMM_WORLD there too

**Comment by streeve** [comment]
@MattRolchigo ready for review

File: src/Finch_Run.hpp [hunk 1]
```diff
@@ -110,13 +110,13 @@ class Layer
 
     auto writeSolidificationData() { return solidification_data_.write(); }
 
-    std::array<double, 3> getLowerSolidificationDataBounds()
+    std::array<double, 3> getLowerSolidificationDataBounds( MPI_Comm comm )
```
Comment (MattRolchigo) [inline src/Finch_Run.hpp:113]: Conflicts with ExaCA/bin/runCoupled.cpp line 53 (which uses a call to this function without input arguments)

File: src/Finch_Run.hpp [hunk 2]
```diff
@@ -110,13 +110,13 @@ class Layer
...
     {
-        return solidification_data_.getLowerBounds();
+        return solidification_data_.getLowerBounds( comm );
     }
-    std::array<double, 3> getUpperSolidificationDataBounds()
+    std::array<double, 3> getUpperSolidificationDataBounds( MPI_Comm comm )
```
Comment (MattRolchigo) [inline src/Finch_Run.hpp:117]: Conflicts with ExaCA/bin/runCoupled.cpp line 54 (which uses a call to this function without input arguments)

**Review by MattRolchigo — COMMENTED** [review]
This will conflict with ExaCA (for FInch-ExaCA coupled runs) - should this be merged simultaneously with an ExaCA pull request that passes the communicator to `getLowerSolidificationBounds` and `getUpperSolidificationBounds`?

File: src/Finch_Run.hpp (hunk 1 above)
Comment (streeve) [reply src/Finch_Run.hpp:126]: Forgot to push my local changes where I marked the previous as deprecated

File: src/Finch_Run.hpp (hunk 2 above)
Comment (streeve) [reply src/Finch_Run.hpp:130]: Fixed

**Review by MattRolchigo — APPROVED** [review]

**Pull request merged**